### DISCUSSION TITLE
PM and ozone verification issues

### DISCUSSION THREAD

**Discussion by edwardstrobach** [dtcenter/METplus#1052] [File I/O] [answered]
Good morning,

Recently I had to redo the July statistics for ozone and pm because we
transitioned to a new operational model.  The layout of the data is the
same and the obs. being used to verify the model output is the same.
However, the pb2nc step only produces empty files for ozone and pm2.5. When looking in the log file I see now errors
(/gpfs/dell2/emc/modeling/noscrub/Edward.Strobach/metplus_aq/CMAQ/logs/master_metplus.log.20210802142753).

I set the LOG_LEVEL to DEBUG in hopes of revealing more information.  The
only thing that kind of stands out is this message here:

DEBUG 3:    Blocking Bufr file (metadata) to:
/gpfs/dell2/emc/modeling/noscrub/Edward.Strobach/metplus_aq/CMAQ/tmp/tmp_pb2nc_tbl_blk_6838_0
DEBUG 1: process_pbfile_metadata() -> the number of records: 61056
DEBUG 1: Processing Bufr File:
 /gpfs/dell1/nco/ops/com/hourly/prod/hourly.20210730/aqm.t12z.prepbufr.tm00
DEBUG 1: Blocking Bufr file to:
/gpfs/dell2/emc/modeling/noscrub/Edward.Strobach/metplus_aq/CMAQ/tmp/tmp_pb2nc_blk_6838_0
DEBUG 3:  PERF: process_pbfile() 0 seconds for preparing
DEBUG 2: Processing 61056 Bufr messages...
DEBUG 4: Request up to 511 vertical levels

Although there is indication that data is being processed, there seems to
be a block on the bufr file?  I don't understand what that means at this
time.  I'm actively trying to debug but am at a loss of what to do.

**Comment by edwardstrobach** [comment, marked as answer]
Update:

I'm not convinced that the problem is resolved quite yet, but when removing the "PB2NC_SKIP_IF_OUTPUT_EXISTS=no" option, I find that it works.  I used this option in the past thinking that it would execute the PB2NC step even if output exists when setting this option to no.

**Reply by JohnHalleyGotway** [reply]
Ed, you do not need to be concerned about this log message:
```
DEBUG 1: Blocking Bufr file to:
/gpfs/dell2/emc/modeling/noscrub/Edward.Strobach/metplus_aq/CMAQ/tmp/tmp_pb2nc_blk_6838_0
```
This is just an implementation detail about reading data from that BUFR file... it just pads the front of the file as needed for reading it with FORTRAN. For earlier versions of MET, we called the NCEP cwordsh utility but moved that extra step into PB2NC instead. And that's the reason for that log message.

Looking in that file, I see no error messages:
```
grep ERROR /gpfs/dell2/emc/modeling/noscrub/Edward.Strobach/metplus_aq/CMAQ/logs/master_metplus.log.20210802142753 | wc -l
> 0
```

I'll check back in on this discussion tomorrow to see if you have any additional issues or Error messages to report.

**Reply by JohnHalleyGotway** [reply]
Ed, this discussion has been open for a long time! Do you have any questions remaining about verifying PM and Ozone with METplus or should I just mark it as being answered?
